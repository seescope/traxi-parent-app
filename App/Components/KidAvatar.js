import React, { PropTypes } from 'react';
import { Image, StyleSheet } from 'react-native';
import * as Animatable from 'react-native-animatable';

import { GOOD, BAD, NEUTRAL, SHADOW_COLOR } from '../Constants/Colours';

const stateColours = {
  good: GOOD,
  bad: BAD,
  neutral: NEUTRAL,
};

const getKidAvatarStyle = (size, state) =>
  StyleSheet.create({
    image: {
      height: size,
      width: size,
      borderRadius: size / 2,
      shadowColor: SHADOW_COLOR,
      shadowOffset: { height: 2 },
      shadowRadius: 5,
      marginBottom: size / 10,
      borderColor: stateColours[state],
      borderWidth: size / 20,
    },
  });

const getAvatarURL = avatarURL => {
  if (avatarURL !== '') return { uri: avatarURL };
  return require('../Images/placeholder_avatar.png');
};

const KidAvatar = ({ avatarURL, size, state, animation, duration, iterationCount }) => (
  <Animatable.Image
    animation={animation}
    iterationCount={iterationCount}
    duration={duration}
    style={getKidAvatarStyle(size, state).image}
    source={getAvatarURL(avatarURL)}
  />
);

KidAvatar.propTypes = {
  avatarURL: PropTypes.string,
  size: PropTypes.number.isRequired,
  state: PropTypes.oneOf(['good', 'bad', 'neutral']),
  animation: PropTypes.string,
  iterationCount: PropTypes.string,
  duration: PropTypes.number,
};

export default KidAvatar;
