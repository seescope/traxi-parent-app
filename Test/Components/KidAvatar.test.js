import React from 'react';
import renderer from 'react-test-renderer';
import KidAvatar from '../../App/Components/KidAvatar';

it('Renders an avatar when an image is supplied', () => {
  const IMAGE_URL = 'http://www4.pictures.gi.zimbio.com/Premiere+MGM+Rocky+Balboa+Arrivals+pczqT_rwQxcl.jpg';
  const TEST_SIZE = 100;
  const tree = renderer.create(<KidAvatar
    avatarURL={IMAGE_URL}
    size={TEST_SIZE}
    state={'neutral'}
  />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Returns a default avatar when none is supplied', () => {
  const TEST_SIZE = 100;
  const tree = renderer.create(<KidAvatar
    avatarURL={''}
    size={TEST_SIZE}
    state={'neutral'}
  />);
  expect(tree.toJSON()).toMatchSnapshot();
});
