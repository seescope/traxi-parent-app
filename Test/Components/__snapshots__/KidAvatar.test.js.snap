exports[`test Renders an avatar when an image is supplied 1`] = `
<Image
  animation={undefined}
  duration={undefined}
  iterationCount={undefined}
  source={
    Object {
      "uri": "http://www4.pictures.gi.zimbio.com/Premiere+MGM+Rocky+Balboa+Arrivals+pczqT_rwQxcl.jpg",
    }
  }
  style={
    Object {
      "borderColor": "rgba(75, 73, 73, 1)",
      "borderRadius": 50,
      "borderWidth": 5,
      "height": 100,
      "marginBottom": 10,
      "shadowColor": "rgba(0,0,0,0.5)",
      "shadowOffset": Object {
        "height": 2,
      },
      "shadowRadius": 5,
      "width": 100,
    }
  } />
`;

exports[`test Returns a default avatar when none is supplied 1`] = `
<Image
  animation={undefined}
  duration={undefined}
  iterationCount={undefined}
  source={1}
  style={
    Object {
      "borderColor": "rgba(75, 73, 73, 1)",
      "borderRadius": 50,
      "borderWidth": 5,
      "height": 100,
      "marginBottom": 10,
      "shadowColor": "rgba(0,0,0,0.5)",
      "shadowOffset": Object {
        "height": 2,
      },
      "shadowRadius": 5,
      "width": 100,
    }
  } />
`;
